### DISCUSSION TITLE
Press back twice to exit?

### DISCUSSION THREAD

**Discussion by mmdevcodes** [react-navigation/react-navigation#9319] [Q&A] [answered]
Is there a recommended way to go about implementing a "press back twice to quit" type functionality? A toast would appear on the first press, and then exit the app if there are no screens left on the navigation stack.

I assume it would be a combination of `useFocusEffect` and `BackHandler`.

**Comment by raajnadar** [comment, marked as answer]
Try using `useState` + `useEffect` + `BackHandler`
